cpp/dolfinx/refinement/utils.cpp

```diff
@@ -652,6 +654,6 @@ mesh::MeshTags<std::int32_t> refinement::transfer_cell_meshtag(
   }
 
   return mesh::MeshTags<std::int32_t>(
-      std::make_shared<mesh::Mesh>(refined_mesh), tdim,
+      std::make_shared<mesh::Mesh>(std::move(refined_mesh)), tdim,
```

Comment:
`move` for `refined_mesh` doesn't make sense here. It won't move, and if it could it would cause a corruption. The old code is wrong too. It makes a copy of the mesh.

The argument `refined_mesh` should probably be a `shared_ptr`, or a more granular design would (i) have a function to get the new `MeshTag` data, then (ii) create the new `MeshTag`.